src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java

```diff
@@ -218,13 +218,18 @@ public void actionPerformed(ActionEvent actionEvent) {
         menu.add(moveToFileDir);
         moveToFileDir.addActionListener(new MoveFileAction(frame, entryEditor, this, true));
 
-        JMenuItem deleteFile = new JMenuItem(Localization.lang("Delete local file"));
+        JMenuItem deleteFile = new JMenuItem(Localization.lang("Permanently delete local file"));
         menu.add(deleteFile);
         deleteFile.addActionListener(e -> {
             int row = getSelectedRow();
+
             // no selection
-            if (row != -1) {
+            if (row == -1) {
+                return;
+            }
 
+            int userConfirm = JOptionPane.showConfirmDialog(null, Localization.lang("Are you sure you want to delete this file?"), Localization.lang("Warning"), JOptionPane.YES_NO_OPTION);
+            if (userConfirm == JOptionPane.YES_OPTION) {
```

Comment:
Hm, one can argue if this really adds so much value. I think the functionality is clear with the question `Do you really want to delete this file?`. It just adds more Strings that need to be translated. However, I'm OK with implementing this. 

In general, what do @JabRef/developers think about this?